### ISSUE TITLE
[BUG]: The controls , searchbar and collapse button should be improved.

### ISSUE THREAD

**Issue by Kuzuri247** [jagpreetrahi/visualize-json-schema#114]
## Description

The dark mode should change the color of controls, the searchbar should be fixed with keyboard accessiblity and proper width and the collpase button icon should be replaced with proper icon.

## Steps to Reproduce

1. Go to website
2. When the mode is toggle the controls color is still white, the searchbar is not properly aligned and collapse button has repetetive svg which is not suitable for the button 

## Expected Behavior

Color of controls should change with mode toggle, searchbar alignment and accessibility should be perfect and collapse button should use proper icon. 

## Screenshots

<img width="506" height="798" alt="Image" src="https://github.com/user-attachments/assets/e8a59ad3-55ef-470c-a356-4a5ecf7e31a6" />

## Are you going to work on this?

- [x] Yes
- [ ] No
